report.md

```diff
@@ -115,11 +115,15 @@ Concerning our issue, since it is related to a new algorithm, there are obviousl
 
 Hence we create the class that will allow us to test the new algorithm of our issue: 'class `TestBitmasking` in the file [test_dp.py](tests/test_dp.py). Each [requirement](TODO) will have its own method that test it inside this class.
 
-## UML for the Bitmasking algorithm
+## UML for the Bitmasking algorithms
 Our solution consists of an algorithm that utilizes several functions. In order to gain a better understanding of the control flow of the algorithm the following control-flow diagram can be consulted. In this figure the boxes represent functions called and the diamonds represents major decision points. The filled dot represents the start and the circled dots represent the termination of the algorithm.
```

Comment:
Could refactor this paragraph to refer to both diagrams? Additionally, could you add a short explanation of why we included control-flow diagrams instead of class diagrams?